tests/Benchmarks/System.Text.Formatting/StringFormatterPerf.cs

```diff
@@ -0,0 +1,241 @@
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                sb.Append(((int)(i % 10)));
+            }
+
+            var text = sb.ToString();
+        }
+
+        [Benchmark]
+        public void InvariantFormatIntDecClr()
+        {
+            var sb = new StringBuilder(NumbersToWrite);
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                sb.Append(((int)(i % 10)));
+            }
+
+            var text = sb.ToString();
+        }
+
+        [Benchmark]
+        public void InvariantFormatIntHex()
+        {
+            var sb = new StringFormatter(NumbersToWrite, pool);
+            var format = new StandardFormat('X', StandardFormat.NoPrecision);
+
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                sb.Append(((int)(i % 10)), format);
+            }
+
+            var text = sb.ToString();
+        }
+
+        [Benchmark]
+        public void InvariantFormatIntHexClr()
+        {
+            var sb = new StringBuilder(NumbersToWrite);
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                sb.Append(((int)(i % 10)).ToString("X"));
+            }
+
+            var text = sb.ToString();
+        }
+
+        [Benchmark]
+        public void InvariantFormatStruct()
+        {
+            var sb = new StringFormatter(NumbersToWrite * 2, pool);
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                sb.Append(new Age(i % 10));
+            }
+
+            var text = sb.ToString();
+        }
+
+        [Benchmark]
+        public void FormatGuid()
+        {
+            var guid = Guid.NewGuid();
+            var guidsToWrite = NumbersToWrite / 10;
+
+            var sb = new StringFormatter(guidsToWrite * 36, pool);
+            for (int i = 0; i < guidsToWrite; i++)
+            {
+                sb.Append(guid);
+            }
+
+            var text = sb.ToString();
+        }
+
+        [Benchmark]
+        public void InvariantFormatStructClr()
+        {
+            var sb = new StringBuilder(NumbersToWrite * 2);
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                sb.Append(new Age(i % 10));
+            }
+
+            var text = sb.ToString();
+        }
+
+        [Benchmark]
+        public void CustomCultureFormat()
+        {
+            var sb = new StringFormatter(NumbersToWrite * 3, pool);
+            sb.SymbolTable = CreateCustomCulture();
+
+            sb.Clear();
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                var next = (i % 128) + 101;
+                sb.Append(next);
+            }
+
+            var text = sb.ToString();
+        }
+
+        [Benchmark]
+        public void CustomCultureFormatClr()
+        {
+            var sb = new StringBuilder(NumbersToWrite * 3);
+            var culture = new CultureInfo("th");
+
+            sb.Clear();
```

Comment:
why do you call the `.Clear` method here? the string builder should be empty right after the creation